### ISSUE TITLE
Equations (2.4) are not obvious from Figure 2.1 as far as I can tell

### ISSUE THREAD

**Issue by chrisoffner3d** [SLAM-Handbook-contributors/slam-handbook-public-release#19]
![figure](https://github.com/user-attachments/assets/c48f5fa0-069a-4ddb-8c21-209b8e65452b)
![figur2](https://github.com/user-attachments/assets/e12276f4-7265-4929-9e0d-1d72cfaeac88)

Sorry, where does measurement $\boldsymbol{z}_4$ in equation (2.4d) come from?

In Figure 2.1 I see three dashed lines corresponding to observations/measurements but I don't know what observation $\boldsymbol{z}_1$ to $\boldsymbol{z}_4$ in the equations correspond to, respectively. From Figure 2.1 it is not clear how many observations $\boldsymbol{z}_i$ there even are – it's only implied by the factorization in equations (2.4a-d).

Only after checking [Factor Graphs for Robot Perception](https://ieeexplore.ieee.org/document/8187520), which is the basis for this chapter, did I find this Bayes net figure below, which illustrates what the observations $\boldsymbol{z}_i$ correspond to. However, this clarifying figure is omitted from the SLAM Handbook, as far as I can tell.

![bayesnet](https://github.com/user-attachments/assets/16812490-de85-4ffa-ad48-ded8f673774d)

**Comment by Shubodh** [comment]
Hey @chrisoffner3d ,  
Check Figure 2.2, equations 2.5a - 2.5d and the text on Pg 25 (and also Pg 24) answers the questions. Take a look at the highlighted text below. 
TL;DR: Measurements Z are intentionally not explicitly shown. But can be done so as explained in the last statement in attached image.

![image](https://github.com/user-attachments/assets/8bf642cb-c4e1-4c3a-9a69-1a722f86bb64)

**Comment by chrisoffner3d** [comment]
The text refers to the measurements not being explicitly shown in the factor graph (i.e. Figure 2.2) but we'd still need to know how many observations $\boldsymbol{z}_i$ we're dealing with (in this case, apparently $i \in \{1, 2, 3, 4\}$), right? But this information is not clear from Figure 1.2 in the SLAM handbook. It his, however, clear from Figure 1.2. in the original "[Factor Graphs for Robot Perception](https://ieeexplore.ieee.org/document/8187520).

**Comment by dellaert** [comment]
I agree it’s not clear, thanks for flagging! I’ll update the text.

**Comment by lucacarlone** [comment]
@chrisoffner3d  feel free to post your full name below such that we can acknowledge your comments in the final draft!

**Comment by chrisoffner3d** [comment]
> @chrisoffner3d  feel free to post your full name below such that we can acknowledge your comments in the final draft!

My name is **Chris Offner**.

Thank you! :)

**Comment by lucacarlone** [comment]
perfect - will do!

**Comment by Chuong98CC** [comment]
For people that are not familiar with factor graph like me, I had a hard time to understand right from the Toy Example in Fig 1.2 (you know, feeling intimidating when you can't understand the Toy example), but from the text book, it is not clear:
- why we have `phi_6` at `p1`, but not `phi_xx`at P2 and P3. 
- Also, why there is a connection between `p1` to `l1`, but not `p2` to `l2`. 
Thanks to @chrisoffner3d , I can trace back to the [Factor Graphs for Robot Perception](https://ieeexplore.ieee.org/document/8187520) to see the figure 1.2 he mentioned.
I think, at the introduction chapter, especially Toy Example, please be more details about the annotation. 
Thank you for a great book.

**Comment by chrisoffner3d** [comment]
I concur with @Chuong98CC in that I found the **Factor Graphs for SLAM** chapter didactically suboptimal. It didn't strike me as a sufficiently self-contained tutorial for someone unfamiliar with factor graphs.

**Comment by lucacarlone** [comment]
here is the new pdf: https://www.dropbox.com/scl/fi/8rkkzkik8wm4dq968md80/main.pdf?rlkey=vtwa62k6iar8ew7e7t2zwa2k1&dl=0

The public repo will also be updated shortly.

 @Chuong98CC feel free to share your full name if you want to be acknowledged in the handbook!
